app/models/container_deployment.rb

```diff
@@ -6,5 +6,395 @@ class ContainerDeployment < ApplicationRecord
   has_many :container_volumes, :as => :parent
   has_many :custom_attributes, :as => :resource, :dependent => :destroy
   has_many :authentications, :as => :resource, :dependent => :destroy
-  serialize :customize, Hash
+  serialize :customizations, Hash
+  AUTHENTICATIONS_TYPES = {:AllowAllPasswordIdentityProvider => AuthenticationAllowAll, :HTPasswdPasswordIdentityProvider => AuthenticationHtpasswd, :LDAPPasswordIdentityProvider => AuthenticationLdap, :RequestHeaderIdentityProvider => AuthenticationRequestHeader, :OpenIDIdentityProvider => AuthenticationOpenId, :GoogleIdentityProvider => AuthenticationGoogle, :GitHubIdentityProvider => AuthenticationGithub, :ssh => AuthPrivateKey, :rhsm => AuthenticationRhsm}.freeze
+  AUTHENTICATIONS_NAMES = {:AllowAllPasswordIdentityProvider => "all", :HTPasswdPasswordIdentityProvider => "htPassword", :LDAPPasswordIdentityProvider => "ldap", :RequestHeaderIdentityProvider => "requestHeader", :OpenIDIdentityProvider => "openId", :GoogleIdentityProvider => "google", :GitHubIdentityProvider => "github", :ssh => "ssh", :rhsm => "rhsm"}.freeze
+
+  def ssh_auth
+    authentications.where(:type => "AuthPrivateKey").first
```

Comment:
`authentications.find_by_type("AuthPrivateKey")` might be nicer.
Same for the queries below.